### ISSUE TITLE
Pathname.elf? and Pathname.dynamic? produce syntax errors if path contains parentheses

### ISSUE THREAD

**Issue by bw-yuhogg** [Linuxbrew/brew#359]
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is a problem with Linuxbrew/brew and not specific formulae? If it's a formulae-specific problem please file this issue at https://github.com/Linuxbrew/homebrew-core/issues/new

```
[bob@bob-Lemur] [~/.linuxbrew/Library/Homebrew] [git:master] brew irb
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
irb(main):001:0> x = Pathname.new("/home/linuxbrew/.linuxbrew/Cellar/cairo/1.14.8/share/gtk-doc/html/cairo/cairo-Quartz-(CGFont)-Fonts.html")
=> #<Pathname:/home/linuxbrew/.linuxbrew/Cellar/cairo/1.14.8/share/gtk-doc/html/cairo/cairo-Quartz-(CGFont)-Fonts.html>
irb(main):002:0> x.elf?
sh: 1: Syntax error: "(" unexpected
=> nil
irb(main):003:0> x.dynamic?
sh: 1: Syntax error: "(" unexpected
=> nil
```

A symptom of this occurs if you run `brew linkage cairo`. Note that this is, in my opinion, not a high priority issue, just pointing it out.

**Comment by sjackman** [comment]
Ah hah! Thank you, Bob! I've seen the error in the logs, and I was wondering what the heck it meant.

**Comment by maxim-belkin** [comment]
🐛

**Comment by maxim-belkin** [comment]
```
$ brew irb
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
irb(main):001:0> x = Pathname.new("/home/linuxbrew/.linuxbrew/Cellar/cairo/1.14.8/share/gtk-doc/html/cairo/cairo-Quartz-\\(CGFont\\)-Fonts.html")
=> #<Pathname:/home/linuxbrew/.linuxbrew/Cellar/cairo/1.14.8/share/gtk-doc/html/cairo/cairo-Quartz-\(CGFont\)-Fonts.html>
irb(main):002:0> x.elf?
=> nil
```

**Comment by maxim-belkin** [comment]
Solution: http://stackoverflow.com/a/7684238/7410631 by @pal

**Comment by maxim-belkin** [comment]
Fixed in #362

**Comment by bw-yuhogg** [comment]
Thanks @maxim-belkin!
